org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -559,7 +559,7 @@ public static void applyEdits(@Nullable IDocument document, @Nullable List<? ext
 
 	@Nullable
 	public static IDocument getDocument(@Nullable IResource resource) {
-		if (resource == null) {
+		if (resource == null || !resource.isAccessible()) {
```

Comment:
To avoid side effects, I propose you move this check to  the `LanguageServerWrapper.underlyingFileMoved`, what do you think?